### PR TITLE
Nicer example plots

### PR DESCRIPTION
**Pull request by mstimberg**
This "beautifies" the example plots. I did not modify the Brunel/Hakim example, but I guess we have plotting code for it somewhere?

Here are the plots as a result of running:
```
$ python cobahh.py --devicename cpp_standalone --scenario uncoupled --N 5000 --monitors --no-profiling
$ python stdp.py --devicename cpp_standalone --delays none --no-profiling --monitors
$ python mushroombody.py --devicename cpp_standalone --N 10000 --no-profiling --monitors
```

Note that I also fixed the difference in scaling of the STDP example with respect to `brian2cuda/tests/features/speed.py` (@denisalevi maybe have a look that I did this correctly).

cobahh_cpp_scenario-uncoupled_N-5000_no-profiling_monitors_no-single-precision_atomics_bundle-mode_vuvuzela

stdp_cpp_delays-none_post-effects_N-10000_no-profiling_monitors_single-precision_atomics_bundle-mode_vuvuzela

mushroombody_cpp_N-10000_no-profiling_monitors_no-single-precision_atomics_bundle-mode_vuvuzela

### CODE REVIEW

**Comment by denisalevi** [comment]
Thanks @mstimberg, looks great!

1. Could you base any changes on the `paper2022-revision` branch? Because master already moved forward with changes that should not belong into the paper code (e.g. changing the code slots, which currently would break the benchmarks). Could you reapply your commits on `paper2022-revision` instead?
1. I will check the STDP example again. But why is it running for 20s? Shouldn't we just run it for 10s and mention that the weight distribution is bimodal after ~100s?
2. The plotting code for the Brunel/Hakim example that I used for figure 1 is [here](https://github.com/denisalevi/brian2cuda-paper/blob/master/talks/2021-05%20Denis%20group%20meeting%20talk/code-snippets/Brunel_Hakim_1999_paperexample_for_talk_with_plots_denis.py). But I guess we don't need it in the supps since its already in figure 1, right?

**Comment by mstimberg** [comment]
> 1. Could you base any changes on the `paper2022-revision` branch? Because master already moved forward with changes that should not belong into the paper code (e.g. changing the code slots, which currently would break the benchmarks). Could you reapply your commits on `paper2022-revision` instead?

Sure, will do.

> 2\. I will check the STDP example again. But why is it running for 20s? Shouldn't we just run it for 10s and mention that the weight distribution is bimodal after ~100s?

It was a bit of a compromise to see the weight distribution changing in the last plot, but happy to run it for 10s (maybe making the runtime a command line argument?)

> 3\. The plotting code for the Brunel/Hakim example that I used for figure 1 is [here](https://github.com/denisalevi/brian2cuda-paper/blob/master/talks/2021-05%20Denis%20group%20meeting%20talk/code-snippets/Brunel_Hakim_1999_paperexample_for_talk_with_plots_denis.py). But I guess we don't need it in the supps since its already in figure 1, right?

I agree, but for consistency it would be nice if the same (or at least a similar) plot gets generated when you run `brunelhakim.py` in the examples.

**Comment by mstimberg** [comment]
All of the points :point_up: should now be fixed.

**Comment by denisalevi** [comment]
I made a few more changes @mstimberg , cleaning up the example scripts.

Relevant for us: 
- I changed the runtime to be the runtime from the examples in Brian2, that means whatever makes sense for the examples (even if that is not what we used in the benchmarks). That is because these scripts are meant for users, not for reviewers. The runtime can be set for all scripts now via `--runtime` as you already implemented for the STDP benchmark.
- I changed the name for the partitioning parameter, so it can be recognized by reviewers.

File: examples/brunelhakim.py [hunk 1]
```diff
@@ -73,25 +73,27 @@
...
-          'num_blocks': num_blocks,
-          'atomics': atomics,
-          'bundle_mode': bundle_mode}
-
+from collections import OrderedDict
 from utils import set_prefs, update_from_command_line
 
-# update params from command line
+# create paramter dictionary that can be modified from command line
+params = OrderedDict([('devicename', devicename),
```
[the remark this answers is not in the document]
Comment (denisalevi) [reply examples/brunelhakim.py:81]: Ah, didn't know that! I was just following a comment in the TODO.txt file, I guess that was pretty old ahaha. I'll change it.

**Comment by denisalevi** [comment]
You are done here @mstimberg, right? I'll merge this.

**Pull request merged**